packages/devtools_app/lib/src/debugger/flutter/debugger_screen.dart

```diff
@@ -88,60 +74,52 @@ class DebuggerScreenBodyState extends State<DebuggerScreenBody>
     final newController = Controllers.of(context).debugger;
     if (newController == controller) return;
     controller = newController;
-    addAutoDisposeListener(controller.isPaused, _onPaused);
-    addAutoDisposeListener(controller.breakpoints);
-  }
 
-  Future<void> _onPaused() async {
-    // TODO(https://github.com/flutter/devtools/issues/1648): Allow choice of
-    // the scripts on the stack.
-    if (controller.isPaused.value) {
-      final currentStack = await controller.getStack();
+    addAutoDisposeListener(controller.currentScript, () {
       setState(() {
-        stack = currentStack;
+        script = controller.currentScript.value;
       });
-      if (stack == null || stack.frames.isEmpty) return;
-      final currentScript =
-          await controller.getScript(stack.frames.first.location.script);
+    });
+    addAutoDisposeListener(controller.currentStack, () {
       setState(() {
-        script = currentScript;
+        stack = controller.currentStack.value;
       });
-    } else {
+    });
+    addAutoDisposeListener(controller.scriptList, () {
       setState(() {
-        stack = null;
+        scriptList = controller.scriptList.value;
       });
-    }
+    });
+    breakpoints = controller.breakpoints.value;
+    addAutoDisposeListener(controller.breakpoints, () {
+      setState(() {
+        breakpoints = controller.breakpoints.value;
+      });
+    });
+
+    controller.getScripts();
   }
 
-  Future<void> onScriptSelected(ScriptRef ref) async {
+  Future<void> _onScriptSelected(ScriptRef ref) async {
     if (ref == null) return;
     setState(() {
       loadingScript = ref;
       script = null;
     });
-    final result = await serviceManager.service.getObject(
-      serviceManager.isolateManager.selectedIsolate.id,
-      ref.id,
-    ) as Script;
-
-    setState(() {
-      script = result;
-    });
+    await controller.selectScript(ref);
   }
 
-  Map<int, Breakpoint> getBreakpointsForLines() {
+  Map<int, Breakpoint> _breakpointsForLines() {
     if (script == null) return {};
-    String getScriptId(Breakpoint b) => b.location.script.id;
-
-    final bps = controller.breakpoints.value
-        .where((b) => b != null && getScriptId(b) == script.id);
+    final bps = breakpoints
```

Comment:
not caused by you but while you are here can you cleanup the `bps` variable name or eliminate it completely by rolling it right into the for loop.